modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/unit_of_measure_selector/UnitOfMeasureSelector.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClaySelectWithOption} from '@clayui/form';
+import {useLiferayState} from '@liferay/frontend-js-state-web';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React, {forwardRef, useCallback, useEffect, useState} from 'react';
+
+import ServiceProvider from '../../ServiceProvider/index';
+import skuOptionsAtom from '../../utilities/atoms/skuOptionsAtom';
+import {
+	CP_INSTANCE_CHANGED,
+	CP_UNIT_OF_MEASURE_SELECTOR_CHANGED,
+} from '../../utilities/eventsDefinitions';
+import {getMinQuantity} from '../../utilities/quantities';
+
+const UnitOfMeasureSelector = forwardRef(
+	({
+		accountId,
+		channelId,
+		cpInstanceId,
+		disabled,
+		name,
+		namespace,
+		productConfiguration,
+		productId,
+		size,
+	}) => {
+		const [inputProperties, setInputProperties] = useState({
+			quantity: getMinQuantity(productConfiguration?.minOrderQuantity, 1),
+			resetQuantity: false,
+			unitOfMeasures: [],
+			value: null,
+		});
+		const [skuId, setSkuId] = useState(cpInstanceId);
+		const [skuOptionsAtomState] = useLiferayState(skuOptionsAtom);
+
+		const DeliveryCatalogAPIServiceProvider = ServiceProvider.DeliveryCatalogAPI(
+			'v1'
+		);
+
+		const postChannelProductSkuBySkuOption = useCallback(
+			(unitOfMeasureKey) => {
+				DeliveryCatalogAPIServiceProvider.postChannelProductSkuBySkuOption(
+					channelId,
+					productId,
+					accountId,
+					inputProperties.quantity,
+					unitOfMeasureKey,
+					skuOptionsAtomState.skuOptions
+				).then((cpInstance) => {
+					cpInstance.skuOptions = skuOptionsAtomState.skuOptions;
+					cpInstance.skuId = parseInt(cpInstance.id, 10);
+
+					const dispatchedPayload = {
+						cpInstance,
+						namespace,
+					};
+
+					Liferay.fire(
+						`${namespace}${CP_INSTANCE_CHANGED}`,
+						dispatchedPayload
+					);
+				});
+			},
+			// eslint-disable-next-line react-hooks/exhaustive-deps
+			[skuOptionsAtomState]
+		);
+
+		useEffect(() => {
+			if (cpInstanceId) {
+				DeliveryCatalogAPIServiceProvider.getChannelProductSku(
+					channelId,
+					productId,
+					cpInstanceId,
+					accountId
+				).then((cpInstance) => {
+					const skuUnitOfMeasures =
+						cpInstance.skuUnitOfMeasures || [];
+
+					setInputProperties((inputProperties) => ({
+						...inputProperties,
+						quantity: getMinQuantity(
+							productConfiguration?.minOrderQuantity,
+							skuUnitOfMeasures[0]?.incrementalOrderQuantity || 1,
+							skuUnitOfMeasures[0]?.precision || 0
+						),
+						resetQuantity: true,
+						unitOfMeasures: skuUnitOfMeasures,
+						value: skuUnitOfMeasures[0]?.key || '',
+					}));
+				});
+			}
+			// eslint-disable-next-line react-hooks/exhaustive-deps
+		}, []);
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Same here, maybe it's another error, it disables the line but not for the dependencies (?)

Reply:
Same as the previous one